charts/app/templates/_helpers.tpl

```diff
@@ -43,5 +43,6 @@ Selector labels
 {{- define "selectorLabels" -}}
 app.kubernetes.io/name: {{ include "fullname" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
+app: {{ .Release.Name }}
```

Comment:
do we need this new label?

Reply:
It's so that this label variant of the cleanup action can be re-used here https://github.com/bcgov/nr-compliance-enforcement/blob/762d12156a2330d9a9f9230e764327e697a85d2d/.github/workflows/pr-close.yml#L23

As the helm pr-close here https://github.com/bcgov/quickstart-openshift-helpers/blob/main/.github/workflows/.pr-close.yml#L161 does not allow flag passthrough. Specificaly in this case `global.secrets.persist=false` as the dev team would like secret workloads to persist in their dev environments.

Reply:
changes to their secret workloads to persist between deployments*

Reply:
lets discuss this, as clean up with label will make helm to stick around with secret objects, @DerekRoberts and I can look at the helper function to add in necessary flags as needed

Reply:
I am sorry.
I misunderstood the change, I see now, that you are cleaning up with helm then with label to make sure the persisted secret is deleted with labels, as helm wont delete them during uninstall

I will keep that into our helper as a backlog item, so we don't do twice cleanups :)

Reply:
@jon-funk I would recommend strongly against persisting DEV secrets, since they tend to be abused.  Keep them GitHub Actions (or even, yuk, Vault), then populate during PR environment creation.

Reply:
@DerekRoberts @mishraomp  - Let me provide a little context as to why we want this change.

Our project has two repositories.  This one, and the nr-compliance-enforcement-cm repository that has the case management APIs and data that live in the emerald cluster.

By default the PRs in this repository are pointing to a static version of dev, in our emerald namespace.

However in the event of PRs that require changes both here and for the other repository we will end up needing to overwrite that secret to point to the environment where the other PR is deployed (e.g. dev-4).

The problem is... EVERY single commit on this repository (e.g. bringing PR up to date with the latest changes) will reset that overridden secret back to dev-static, which will usually result in integration tests failing.  

As a result whenever changes to a PR are required one needs to
- Commit Changes
- Revert secret from default value back to overridden value
- Ensure that initial helm deployment has finished
- Nudge OpenShift pods so that proper secret is picked up
- Restart integration tests if they were failing because earlier steps weren't done in time

Can you provide any suggestions for how we can streamline this that doesn't involve persisting secrets?

Reply:
> ent of PRs that require changes both here and for the other repository we will end up needing to overwrite that secret to point to the environment where the other PR is deployed (e.g. dev-4).

lets have a quick discussion on this one over teams, we can have the .deployer.yml in the repo itself and pass in github secrets, which would allow us github secrets as the source of truth , so we can keep a human generated secret in both repos, which can be fed to the pipeline and we don't need to persist secrets across prs or rollouts.

Reply:
@jon-funk , lets update the logic here to look for pr specific secret objects, if they exist, read from them, if not read from default, https://github.com/bcgov/nr-compliance-enforcement/blob/main/charts/app/templates/secret.yaml#L5

Happy to have a chat on this over teams

Reply:
@afwilcox @jon-funk  How about reading secrets from the second repo and synching them over?  It'd eliminate a lot of busy work.  There are tons of alternatives too, like just plain copying the remote repo's config to GitHub secrets to easily (re-)population.  @mishraomp's chat idea works too, since this isn't as easy as we usually see.  ;)